src/uk/gov/hmcts/contino/GithubAPI.groovy

```diff
@@ -49,16 +52,30 @@ class GithubAPI {
       requestBody: "${body}",
       consoleLogResponseBody: true,
       validResponseCodes: '200')
+
+    this.cachedLabelList = []
+  }
+
+/**
+ * Check this.cachedLabelList, if empty, call getLabels() to repopulate.
+*/
+  private getLabelsFromCache() {
+    if (this.cachedLabelList.size() == 0) {
+      def project = currentProject()
+      def pullRequestNumber = currentPullRequestNumber()
+      return getLabels(project, pullRequestNumber)
+    }
+
+    return this.cachedLabelList
   }
 
+
 /**
  * Check Pull Request for label by a pattern in name.
  */
   def getLabelsbyPattern(String branch_name, String key) {
     if (new ProjectBranch(branch_name).isPR() == true) {
-      def project = currentProject()
-      def pullRequestNumber = currentPullRequestNumber()
-      return getLabels(project, pullRequestNumber).findAll{it.contains(key)}
+      return this.getLabelsFromCache().findAll{it.contains(key)}
```

Comment:
is there a reason you didn't change `getLabels` internally? Given it's part of the public API it would be good if it was transitioned to use the cache as well for safety I think

Reply:
there shouldn't be any need to update the cache, and if we need to we can add it I think.

Reply:
getLabels() now utilises the cache.  Added refreshLabelCache() which can be called to update the cache if required.